### PR TITLE
Identity V3: add Trust Get and List funcions

### PR DESCRIPTION
**Pull request by Pirionfr**
Implement Get and List functions for v3/extensions/trusts package
with unit tests and docs.

For #1642 

Links to the line numbers/files in the OpenStack source code that support the
code in this PR:

https://github.com/openstack/keystone/blob/stable/stein/keystone/trust/core.py#L118
https://github.com/openstack/keystone/blob/stable/stein/keystone/api/trusts.py#L197
https://github.com/openstack/keystone/blob/stable/stein/keystone/api/trusts.py#L156

### CODE REVIEW

File: openstack/identity/v3/extensions/trusts/testing/fixtures.go [hunk 1]
```diff
@@ -5,6 +5,8 @@ import (
 	"net/http"
 	"testing"
 
+	"github.com/gophercloud/gophercloud/openstack/identity/v3/extensions/trusts"
+
```
Comment (ozerovandrei) [inline openstack/identity/v3/extensions/trusts/testing/fixtures.go]: Redundant empty line.

File: openstack/identity/v3/extensions/trusts/testing/requests_test.go [hunk 2]
```diff
@@ -4,6 +4,8 @@ import (
 	"testing"
 	"time"
 
+	"github.com/gophercloud/gophercloud/pagination"
+
```
Comment (ozerovandrei) [inline openstack/identity/v3/extensions/trusts/testing/requests_test.go]: Redundant empty line.

**Comment by ozerovandrei** [comment]
Oh I forgot about a browser tab with this PR 😄 
Well those redundant empty lines in imports can be fixed in later PRs.

File: openstack/identity/v3/extensions/trusts/testing/fixtures.go (hunk 1 above)
Comment (jtopjian) [reply openstack/identity/v3/extensions/trusts/testing/fixtures.go]: Good catch - I'll fix this in #1857

**Pull request merged**